var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -394,6 +394,7 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     patch('fix_clang_errors_12_18_1.patch', when='@12.18.1%clang')
     patch('cray_secas_12_12_1.patch', when='@12.12.1%cce')
     patch('cray_secas.patch', when='@12.14.1:12%cce')
+    patch('fix_teuchos_complex_cast.patch', when='@:13.2.0')
```

Comment:
It's better to reference an existing patch directly (with checksum) than to distribute the patch with spack:
```suggestion
    patch(url='https://patch-diff.githubusercontent.com/raw/trilinos/Trilinos/pull/10545.patch', when='@:13.2.0 +complex')
```
This also adds the `+complex` variant since I assume it doesn't affect downstream code when complex is disabled.

Reply:
Sure, the `+complex` would be the only time it would need to call things that are patched. Thanks for the suggestion about referring to the online resource for the patch.